### ISSUE TITLE
pkg_resources.ContextualVersionConflict

### ISSUE THREAD

**Issue by odevel42** [sphinx-doc/sphinx#11641]
### Describe the bug

I get the following error message with the command "make html":
```
# Platform:         linux; (Linux-4.18.0-372.32.1.el8_6.x86_64-x86_64-with-glibc2.31)
# Sphinx version:   7.2.3
# Python version:   3.9.16 (CPython)
# Docutils version: 0.20.1
# Jinja2 version:   3.1.2
# Pygments version: 2.15.1

# Last messages:


# Loaded extensions:


# Traceback:
Traceback (most recent call last):
  File "/opt/conda/envs/py310/lib/python3.9/site-packages/sphinx/cmd/build.py", line 293, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
  File "/opt/conda/envs/py310/lib/python3.9/site-packages/sphinx/application.py", line 263, in __init__
    self.env = self._init_env(freshenv)
  File "/opt/conda/envs/py310/lib/python3.9/site-packages/sphinx/application.py", line 305, in _init_env
    return self._create_fresh_env()
  File "/opt/conda/envs/py310/lib/python3.9/site-packages/sphinx/application.py", line 310, in _create_fresh_env
    env = BuildEnvironment(self)
  File "/opt/conda/envs/py310/lib/python3.9/site-packages/sphinx/environment/__init__.py", line 262, in __init__
    self.setup(app)
  File "/opt/conda/envs/py310/lib/python3.9/site-packages/sphinx/environment/__init__.py", line 292, in setup
    for domain in app.registry.create_domains(self):
  File "/opt/conda/envs/py310/lib/python3.9/site-packages/sphinx/registry.py", line 173, in create_domains
    domain = DomainClass(env)
  File "/opt/conda/envs/py310/lib/python3.9/site-packages/sphinxcontrib/bibtex/domain.py", line 289, in __init__
    self.reference_style = style()
  File "<string>", line 10, in __init__
  File "<string>", line 9, in __init__
  File "/opt/conda/envs/py310/lib/python3.9/site-packages/sphinxcontrib/bibtex/style/referencing/__init__.py", line 154, in __post_init__
    self.style_plugin = pybtex.plugin.find_plugin(
  File "/opt/conda/envs/py310/lib/python3.9/site-packages/pybtex/plugin/__init__.py", line 106, in find_plugin
    return _load_entry_point(plugin_group, name, use_aliases=True)
  File "/opt/conda/envs/py310/lib/python3.9/site-packages/pybtex/plugin/__init__.py", line 80, in _load_entry_point
    return entry_point.load()
  File "/opt/conda/envs/py310/lib/python3.9/site-packages/pkg_resources/__init__.py", line 2467, in load
    self.require(*args, **kwargs)
  File "/opt/conda/envs/py310/lib/python3.9/site-packages/pkg_resources/__init__.py", line 2490, in require
    items = working_set.resolve(reqs, env, installer, extras=self.extras)
  File "/opt/conda/envs/py310/lib/python3.9/site-packages/pkg_resources/__init__.py", line 815, in resolve
    dist = self._resolve_dist(
  File "/opt/conda/envs/py310/lib/python3.9/site-packages/pkg_resources/__init__.py", line 861, in _resolve_dist
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (MarkupSafe 1.1.1 (/home/kubeflow/.conda-persistent/envs/py310/lib/python3.9/site-packages), Requirement.parse('MarkupSafe>=2.0'), {'Jinja2'})
```

### How to Reproduce

make html 
(see error above)

### Environment Information

```text
linux (see above)
```

**Comment by picnixz** [comment]
It's hard to parse the error but I think it's an issue with `pybtex` which seems to be used by `sphinxcontrib-bibtex`. Try with the extension disabled and feel free to re-open the issue if it persists.
